### PR TITLE
Hib 102 rtc snap glue code

### PR DESCRIPTION
**Pull request by williamh890**
Wrapper code for hyp3 processing

## Steps
- [x] Setup working directory
- [x] Package output files from globs
- [x] Upload zip+browse to s3
- [x] Return product link+browse img

## Testing
- [x] Full rtc snap runthrough

## Next (Actual hyp3-in-a-box integration)
- Integration with ec2/worker
- Get earthdata credentials from somewhere for granule download
- Get s3 bucket for products upload
- pre-sign urls
- Separation of library and processes-specific code
- Pass on hyp3 job data like user id, subscription, ... (above separation will help with this.)

### CODE REVIEW

File: processes/rtc_snap/src/working_directory.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import contextlib
+import pathlib as pl
+import shutil
+
+
+@contextlib.contextmanager
+def create(granule):
```
Comment (Askaholic) [inline processes/rtc_snap/src/working_directory.py]: rename to something like `granule_directory`. A little misleading that it's called "create" and yet it actually tears something down

File: processes/rtc_snap/src/rtc_snap.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+
+def process(granule_name, working_dir):
+    subprocess.check_call([
+        'python2', script_path(),
```
Comment (Askaholic) [inline processes/rtc_snap/src/rtc_snap.py]: Maybe just execute the script, and make them include a `#!` so it can be written in any language.

Comment (Askaholic) [reply processes/rtc_snap/src/rtc_snap.py]: In which case it's probably better to pass in all arguments by position, to make CLI parsing as easy as possible

File: processes/rtc_snap/src/package.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        make_ordered_set_from(path)
+        for path in [work_dir_path, output_file_path]
+    ]
+
+    path_without_working_dir = work_dir_set ^ output_file_set
+
+    return pl.Path(*path_without_working_dir)
+
+
+def make_ordered_set_from(path):
```
Comment (Askaholic) [inline processes/rtc_snap/src/package.py]: Does this function really need to exist?

File: processes/rtc_snap/test/rtc_snap_strategies.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+from hypothesis import strategies as st
+
+import asf_granule_util as gu
+
+
+def granules():
```
Comment (Askaholic) [inline processes/rtc_snap/test/rtc_snap_strategies.py:6]: Just a note that this is duplicated in hyp3_event tests. Possible refactor if it comes up again, or there are other duplicate hypothesis strategies.

File: processes/rtc_snap/test/test_working_directory.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    with working_directory.create(granule) as wd:
+        assert granule.unique_id in wd
+        assert 'job' in wd
+
+    make_mock.assert_called_once()
+    teardown_mock.assert_called_once()
+
+
+@pytest.fixture()
+def granule():
```
Comment (Askaholic) [inline processes/rtc_snap/test/test_working_directory.py:21]: `make_test_granule`? `get_test_granule`? `make_example_granule`? What does the function `granule` really do? granule isn't a verb.

Comment (williamh890) [reply processes/rtc_snap/test/test_working_directory.py:21]: Thats how it's done in the pytest fixtures docs.

https://docs.pytest.org/en/latest/fixture.html

File: processes/rtc_snap/src/working_directory.py (hunk 1 above)
Comment (williamh890) [reply processes/rtc_snap/src/working_directory.py]: I was sticking with the naming convention (`open`, `socket.create_connection`, `urllib.request.urlopen`)

File: processes/rtc_snap/src/package.py (hunk 3 above)
Comment (williamh890) [reply processes/rtc_snap/src/package.py]: This is to wrap ordered set, which is a 3rd party library.

Comment (williamh890) [reply processes/rtc_snap/src/package.py]: I was surprised it wasn't in the standard library actually.

File: processes/rtc_snap/test/rtc_snap_strategies.py (hunk 4 above)
Comment (williamh890) [reply processes/rtc_snap/test/rtc_snap_strategies.py:6]: Agreed.

File: processes/rtc_snap/test/test_working_directory.py (hunk 5 above)
Comment (Askaholic) [reply processes/rtc_snap/test/test_working_directory.py:21]: Oh so it needs to have the same name as the argument which gets passed in

File: processes/rtc_snap/src/package.py (hunk 3 above)
Comment (Askaholic) [reply processes/rtc_snap/src/package.py]: So the thought is if you want to change the implementation of which OrderedSet object is used in the future?

Comment (williamh890) [reply processes/rtc_snap/src/package.py]: It's in a very isolated place, that's the thought.

**Pull request merged**